src/components/MobileLogIn.tsx

```diff
@@ -0,0 +1,396 @@
+import { Button, ButtonSize, ButtonType } from "../components/Button";
+import { Link } from "../components/Link";
+import { AppState } from "../reducers";
+import { getLoggedInMember } from "../selectors/auth";
+import { sendAppInstallText } from "../actions";
+import { Loading } from "./Loading";
+import { Member } from "../reducers/membersNew";
+// tslint:disable-next-line:no-var-requires
+const CONFIG = require("../data/config.json");
+
+interface OwnProps {
+  signInSuccessCallback?: () => boolean;
+}
+
+interface StateProps {
+  loggedInMember?: Member;
+  authIsLoaded: boolean;
+}
+
+interface DispatchProps {
+  sendAppInstallText: typeof sendAppInstallText;
+}
+
+type Props = OwnProps & StateProps & DispatchProps;
+
+interface State {
+  mobileNumber: string;
+  confirmationCode: string;
+
+  submittingPhoneNumber: boolean;
+  waitingForRecaptchaVerification: boolean;
+  submittingConfirmationCode: boolean;
+  waitingForConfirmation: boolean;
+  transitionSuccessful: boolean;
+
+  phoneNumberError?: string;
+  confirmationCodeError?: string;
+}
+
+class MobileLogInComponent extends React.Component<Props, State> {
+  private recaptchaContainer: any;
+  private recaptchaVerifier: firebase.auth.RecaptchaVerifier | undefined;
+  private phoneNumberConfirmationResult:
+    | firebase.auth.ConfirmationResult
+    | undefined;
+  // We don't need this state to trigger a component re-render
+  private formattedMobileNumber: string | undefined;
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      mobileNumber: "",
+      confirmationCode: "",
+      submittingPhoneNumber: false,
+      waitingForRecaptchaVerification: true,
+      submittingConfirmationCode: false,
+      waitingForConfirmation: false,
+      transitionSuccessful: false
+    };
+    this._handleLoggedIn();
+  }
+
+  private _handleLoggedIn() {
+    if (this.props.loggedInMember && this.props.signInSuccessCallback) {
+      this.props.signInSuccessCallback();
+    }
+  }
+
+  /**
+   * The recaptcha container may not have been rendered if the user is not
+   * properly authenticated, so we need to check that the recaptcha container
+   * both exists and has not already been initialized before initializing it.
+   */
+  private _initializeRecaptchaContainer() {
+    if (this.recaptchaContainer && !this.recaptchaVerifier) {
+      this.recaptchaVerifier = new firebase.auth.RecaptchaVerifier(
+        "recaptchaContainer"
+      );
+    }
+  }
+
+  public componentDidMount() {
+    this._initializeRecaptchaContainer();
+  }
+
+  public componentDidUpdate(prevProps: Props) {
+    this._initializeRecaptchaContainer();
+    if (this.props.loggedInMember) {
+      this._handleLoggedIn();
+    }
+  }
+
+  public componentWillUnmount() {
+    if (this.recaptchaVerifier) {
+      this.recaptchaVerifier.clear();
+    }
+  }
+
+  private updateMobileNumber = (value: string) => {
+    const formatter = new AsYouTypeFormatter("US");
```

Comment:
What will this mean for non-US people?

Reply:
hmm, I think it will mean that their phone number might be formatted a bit funkily. That being said, the autoformatting isn't so great anyway, so I'll create an issue to look at how Omar's doing in in our app and use that, since that's pretty nice.